version: '3'
services:
  laravel.test:
    build:
      context: ./vendor/laravel/sail/runtimes/8.2
      dockerfile: Dockerfile
      args:
        WWWGROUP: '${WWWGROUP}'
    image: sail-8.2/app
    extra_hosts:
      - 'host.docker.internal:host-gateway'
    ports:
      - "8080:80"      
    environment:
      WWWUSER: '${WWWUSER}'
      LARAVEL_SAIL: 1
      XDEBUG_MODE: '${SAIL_XDEBUG_MODE:-off}'
      XDEBUG_CONFIG: '${SAIL_XDEBUG_CONFIG:-client_host=host.docker.internal}'
      IGNITION_LOCAL_SITES_PATH: './vendor/laravel/sail/runtimes/8.2'
      LARAVEL_WEBSOCKETS_HOST: '0.0.0.0'
      PUSHER_HOST: 'laravel.test'
      PUSHER_PORT: '6001'
      APP_URL: 'http://localhost:8080'
    volumes:
      - '.:/var/www/html'
    networks:
      - sail
    depends_on:
      mysql:
        condition: service_healthy

  mysql:
    image: 'mysql/mysql-server:8.0'
    ports:
      - '3307:3306'    # Puerto mapeado para acceso externo
    environment:
      MYSQL_ROOT_PASSWORD: '${DB_PASSWORD}'
      MYSQL_DATABASE: '${DB_DATABASE}'
      MYSQL_USER: '${DB_USERNAME}'
      MYSQL_PASSWORD: '${DB_PASSWORD}'
    volumes:
      - 'sail-mysql:/var/lib/mysql'
      - './vendor/laravel/sail/database/mysql/create-testing-database.sh:/docker-entrypoint-initdb.d/10-create-testing-database.sh'
    networks:
      - sail
    healthcheck:
      test:
        - CMD
        - mysqladmin
        - ping
        - '-p${DB_PASSWORD}'
      retries: 3
      timeout: 5s
      start_period: 20s
  

networks:
  sail:
    driver: bridge

volumes:
  sail-mysql:
    driver: local